src/lib/components/SettingsModal.svelte

```diff
@@ -10,56 +10,94 @@
 
+	let isConfirmingDeletion = false;
+
 	const dispatch = createEventDispatcher<{ close: void }>();
 </script>
 
 <Modal on:close>
-	<form
-		class="flex w-full flex-col gap-5 p-6"
-		use:enhance={() => {
-			dispatch("close");
-		}}
-		method="post"
-		action="{base}/settings"
-	>
+	<div class="flex w-full flex-col gap-5 p-6">
 		<div class="flex items-start justify-between text-xl font-semibold text-gray-800">
 			<h2>Settings</h2>
 			<button type="button" class="group" on:click={() => dispatch("close")}>
 				<CarbonClose class="text-gray-900 group-hover:text-gray-500" />
 			</button>
 		</div>
+		<form
+			class="flex flex-col gap-5"
+			use:enhance={() => {
+				dispatch("close");
+			}}
+			method="post"
+			action="{base}/settings"
+		>
+			<label class="flex cursor-pointer select-none items-center gap-2 text-gray-500">
+				{#each Object.entries(settings).filter(([k]) => k !== "shareConversationsWithModelAuthors") as [key, val]}
+					<input type="hidden" name={key} value={val} />
+				{/each}
+				<Switch
+					name="shareConversationsWithModelAuthors"
+					bind:checked={settings.shareConversationsWithModelAuthors}
+				/>
+				Share conversations with model authors
+			</label>
 
-		<label class="flex cursor-pointer select-none items-center gap-2 text-gray-500">
-			{#each Object.entries(settings).filter(([k]) => k !== "shareConversationsWithModelAuthors") as [key, val]}
-				<input type="hidden" name={key} value={val} />
-			{/each}
-			<Switch
-				name="shareConversationsWithModelAuthors"
-				bind:checked={settings.shareConversationsWithModelAuthors}
-			/>
-			Share conversations with model authors
-		</label>
-
-		<p class="text-gray-800">
-			Sharing your data will help improve the training data and make open models better over time.
-		</p>
-		<p class="text-gray-800">
-			You can change this setting at any time, it applies to all your conversations.
-		</p>
-		<p class="text-gray-800">
-			Read more about this model's authors,
-			<a
-				href="https://open-assistant.io/"
-				target="_blank"
-				rel="noreferrer"
-				class="underline decoration-gray-300 hover:decoration-gray-700">Open Assistant</a
-			>.
-		</p>
+			<p class="text-gray-800">
+				Sharing your data will help improve the training data and make open models better over time.
+			</p>
+			<p class="text-gray-800">
+				You can change this setting at any time, it applies to all your conversations.
+			</p>
+			<p class="text-gray-800">
+				Read more about this model's authors,
+				<a
+					href="https://open-assistant.io/"
+					target="_blank"
+					rel="noreferrer"
+					class="underline decoration-gray-300 hover:decoration-gray-700">Open Assistant</a
+				>.
+			</p>
+			<button
+				type="submit"
+				class="mt-2 rounded-full bg-black px-5 py-2 text-lg font-semibold text-gray-100 ring-gray-400 ring-offset-1 transition-all focus-visible:outline-none focus-visible:ring hover:ring"
+			>
+				Apply
+			</button>
+		</form>
```

Comment:
Nit: we can still have a global form two ways:

- Adding `formaction` prop to a specific submit button => will change the action of the form when clicking the button
- Adding a `form` prop to an input => will attach the input to the form, even if it's located in another form or outside any forms

It can also help go towards the no-js route - I know that settings can only be opened with JS at the moment, but we could also switch it to `href="?settings"` later-on for people w/o JS